packages/fury-adapter-oas3-parser/lib/parser/oas/parseOauthFlowsObject.js

```diff
@@ -0,0 +1,81 @@
+const R = require('ramda');
+const {
+  isAnnotation, isExtension, hasKey, getValue,
+} = require('../../predicates');
+const {
+  createWarning,
+  createInvalidMemberWarning,
+} = require('../annotations');
+const pipeParseResult = require('../../pipeParseResult');
+const parseObject = require('../parseObject');
+const parseOauthFlowObject = require('./parseOauthFlowObject');
+
+const name = 'Oauth Flows Object';
+
+const isValidFlow = R.anyPass(R.map(hasKey, ['implicit', 'password', 'clientCredentials', 'authorizationCode']));
+const grantTypes = {
+  implicit: 'implicit',
+  password: 'resource owner password credentials',
+  clientCredentials: 'client credentials',
+  authorizationCode: 'authorization code',
+};
+
+/**
+ * Parse Oauth Flows Object
+ *
+ * @param namespace {Namespace}
+ * @param element {Element}
+ * @returns ParseResult
+ *
+ * @see https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#oauthFlowsObject
+ * @private
+ */
+function parseOauthFlowsObject(context, object) {
+  const { namespace } = context;
+
+  const parseFlow = (member) => {
+    const key = member.key.toValue();
+
+    const needAuthorizationUrl = flow => R.includes(key, ['implicit', 'authorizationCode']);
+    const needTokenUrl = flow => R.includes(key, ['password', 'clientCredentials', 'authorizationCode']);
```

Comment:
Is there a way to avoid these unnecessary flow arguments here?

Reply:
```suggestion
    const needTokenUrl = () => R.includes(key, ['password', 'clientCredentials', 'authorizationCode']);
```